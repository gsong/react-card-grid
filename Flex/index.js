import React from "react";

import { flexCompensate } from "../utils";

const Container = React.forwardRef(
  ({ columnGap, rowGap, children, ...props }, ref) => {
    const style = flexCompensate.container({ columnGap, rowGap });

    return (
      <div {...props}>
        <ul data-supple-flex-container style={style} ref={ref}>
          {children}
        </ul>
      </div>
    );
  },
);

const Item = ({ itemWidth, columnGap, rowGap, ...props }) => {
  const style = flexCompensate.item({ itemWidth, columnGap, rowGap });

  return <li data-supple-flex-item style={style} {...props} />;
};

export default { Container, Item };
