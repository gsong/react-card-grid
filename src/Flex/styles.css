[data-supple-flex-container] {
  display: flex;
  flex-wrap: wrap;
  justify-content: flex-start;
  padding: 0;
}

[data-supple-flex-item] {
  list-style: none;
}
