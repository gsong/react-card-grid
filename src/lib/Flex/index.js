//@flow strict
import React from "react";

import { flexCompensate } from "../utils.js";

import type { Gaps, Props } from "../types.js";

const Container = React.forwardRef<Props, ?HTMLElement>(
  ({ columnGap, rowGap, children, ...props }, ref) => {
    const style = flexCompensate.container({ columnGap, rowGap });

    return (
      <div {...props}>
        <ul data-react-listitem-flex-container style={style} ref={ref}>
          {children}
        </ul>
      </div>
    );
  },
);

type ItemProps = { ...Gaps, itemWidth: number, ... };

const Item = ({ columnGap, itemWidth, rowGap, ...props }: ItemProps) => {
  const style = {
    width: itemWidth,
    ...flexCompensate.item({ columnGap, rowGap }),
  };

  return <li data-react-listitem-flex-item style={style} {...(props: any)} />;
};

export default { Container, Item };
