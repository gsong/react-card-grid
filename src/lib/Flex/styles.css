[data-react-listitem-flex-container] {
  display: flex;
  flex-wrap: wrap;
  justify-content: flex-start;
  padding: 0;
}

[data-react-listitem-flex-item] {
  list-style: none;
}
