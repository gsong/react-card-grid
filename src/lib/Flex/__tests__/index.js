//@flow strict
import React from "react";
import { render } from "@testing-library/react";

import Flex from "..";

const children = "fixed-snapshot-children";

describe("Flex.Container", () => {
  test("renders properly", () => {
    const { container } = render(<Flex.Container>{children}</Flex.Container>);
    expect(container).toMatchSnapshot();
  });

  test("renders properly with optional params", () => {
    const { container } = render(
      <Flex.Container columnGap={16} rowGap={8}>
        {children}
      </Flex.Container>,
    );
    expect(container).toMatchSnapshot();
  });
});

describe("Flex.Item", () => {
  test("renders properly", () => {
    const { container } = render(
      <Flex.Item itemWidth={150}>{children}</Flex.Item>,
    );
    expect(container).toMatchSnapshot();
  });

  test("renders properly with optional params", () => {
    const { container } = render(
      <Flex.Item itemWidth={150} columnGap={16} rowGap={8}>
        {children}
      </Flex.Item>,
    );
    expect(container).toMatchSnapshot();
  });
});
