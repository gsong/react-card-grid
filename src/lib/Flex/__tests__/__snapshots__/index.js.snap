// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Flex.Container renders properly 1`] = `
<div>
  <div>
    <ul
      data-react-listitem-flex-container="true"
      style="margin: -0px 0px 0px -0px;"
    >
      fixed-snapshot-children
    </ul>
  </div>
</div>
`;

exports[`Flex.Container renders properly with optional params 1`] = `
<div>
  <div>
    <ul
      data-react-listitem-flex-container="true"
      style="margin: -8px 0px 0px -16px;"
    >
      fixed-snapshot-children
    </ul>
  </div>
</div>
`;

exports[`Flex.Item renders properly 1`] = `
<div>
  <li
    data-react-listitem-flex-item="true"
    style="width: 150px; margin: 0px 0px 0px 0px;"
  >
    fixed-snapshot-children
  </li>
</div>
`;

exports[`Flex.Item renders properly with optional params 1`] = `
<div>
  <li
    data-react-listitem-flex-item="true"
    style="width: 150px; margin: 8px 0px 0px 16px;"
  >
    fixed-snapshot-children
  </li>
</div>
`;
